### PR TITLE
pgcli: 2.0.0 -> 2.0.1

### PR DESCRIPTION
**Pull request by aneeshusa**
###### Motivation for this change
Changelog: https://github.com/dbcli/pgcli/blob/master/changelog.rst#201
Major feature of note is Python 3.7 support.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by worldofpeace** [comment]
@aneeshusa Build fails with:
```
  Could not find a version that satisfies the requirement prompt-toolkit<2.1.0,>=2.0.6 (from pgcli==2.0.1) (from versions: )
No matching distribution found for prompt-toolkit<2.1.0,>=2.0.6 (from pgcli==2.0.1)
```

**Comment by aneeshusa** [comment]
@worldofpeace looks like there was an issue due to nixpkgs still being python2 by default, I have python3 by default and had no issues. Pushed a commit to pin to Python 3 which I've tested locally.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -8807,7 +8807,7 @@ with pkgs;
 
   peg = callPackage ../development/tools/parsing/peg { };
 
-  pgcli = callPackage ../development/tools/database/pgcli {};
+  pgcli = callPackage ../development/tools/database/pgcli { pythonPackages = python3Packages; };
```
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix]: Please just use `python3Packages` within the derivation.

We'd like it to always appear that `python` proper is always python2.

**Comment by worldofpeace** [comment]
But this leaves the python 2 build broken.

If I'm understanding this correctly, applications for both interpreter versions want to migrate to prompt_toolkit v2+. Yet some had no plans to.

It seems things were correct and allowed migration until https://github.com/NixOS/nixpkgs/commit/f0b49c4d10c86497e04a2b707134a8d7e2198783#diff-2afbba83e1b14f9e0c304c769831a0e2 hit.

@FRidh I feel like that should be reverted.

**Comment by aneeshusa** [comment]
Since this is an application and not a library, I think it's fine to not have a Python 2 build. Personally, I'm also not interested in spending time on Python 2 support since it's 2018.

Agreed that https://github.com/NixOS/nixpkgs/commit/f0b49c4d10c86497e04a2b707134a8d7e2198783 seems strange and I'd want to revert it, but maybe there was a good reason for it.

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build pgcli

**Comment by worldofpeace** [comment]
I get this traceback when trying to execute on NixOS:

<details>
<summary>Click to expand</summary>

```
Traceback (most recent call last):
  File "/nix/store/3dqdjjh1n521q2qmrnah6graa5p5m1ay-pgcli-2.0.1/bin/.pgcli-wrapped", line 12, in <module>
    sys.exit(cli())
  File "/nix/store/przda61d8fq6dgw4l3bqb05ica0jmibg-python3.7-click-6.7/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/nix/store/przda61d8fq6dgw4l3bqb05ica0jmibg-python3.7-click-6.7/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/nix/store/przda61d8fq6dgw4l3bqb05ica0jmibg-python3.7-click-6.7/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/nix/store/przda61d8fq6dgw4l3bqb05ica0jmibg-python3.7-click-6.7/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/nix/store/3dqdjjh1n521q2qmrnah6graa5p5m1ay-pgcli-2.0.1/lib/python3.7/site-packages/pgcli/main.py", line 1046, in cli
    pgcli.connect(database, host, user, port)
  File "/nix/store/3dqdjjh1n521q2qmrnah6graa5p5m1ay-pgcli-2.0.1/lib/python3.7/site-packages/pgcli/main.py", line 434, in connect
    passwd = keyring.get_password('pgcli', key)
  File "/nix/store/50h6i4gf9s7ai27a3wavbjpvq8ihljda-python3.7-keyring-13.2.1/lib/python3.7/site-packages/keyring/core.py", line 41, in get_password
    return _keyring_backend.get_password(service_name, username)
  File "/nix/store/50h6i4gf9s7ai27a3wavbjpvq8ihljda-python3.7-keyring-13.2.1/lib/python3.7/site-packages/keyring/backends/SecretService.py", line 67, in get_password
    collection = self.get_preferred_collection()
  File "/nix/store/50h6i4gf9s7ai27a3wavbjpvq8ihljda-python3.7-keyring-13.2.1/lib/python3.7/site-packages/keyring/backends/SecretService.py", line 59, in get_preferred_collection
    collection.unlock()
  File "/nix/store/ajz7wmb8xpd1dqnw8xprf6322r1mqlab-python3.7-secretstorage-2.3.1/lib/python3.7/site-packages/secretstorage/collection.py", line 63, in unlock
    return unlock_objects(self.bus, [self.collection_path], callback)
  File "/nix/store/ajz7wmb8xpd1dqnw8xprf6322r1mqlab-python3.7-secretstorage-2.3.1/lib/python3.7/site-packages/secretstorage/util.py", line 168, in unlock_objects
    return exec_prompt_glib(bus, prompt)[0]
  File "/nix/store/ajz7wmb8xpd1dqnw8xprf6322r1mqlab-python3.7-secretstorage-2.3.1/lib/python3.7/site-packages/secretstorage/util.py", line 127, in exec_prompt_glib
    from gi.repository import GLib
ModuleNotFoundError: No module named 'gi'
```
</details>

**Review by jokogr — APPROVED** [review]
I agree with having pgcli in Python 3 only. It works here...

**Comment by jokogr** [comment]
@worldofpeace could you write how you run the command so that we could reproduce the traceback?

**Comment by worldofpeace** [comment]
> @worldofpeace could you write how you run the command so that we could reproduce the traceback?

I simply executed it with no arguments.

**Comment by Mic92** [comment]
`secretstorage` should depend `pygobject3` to solve the issue worldofpeace has.

**Comment by worldofpeace** [comment]
Merged in 3dbbd9f8c221949e442a4803981cd6d1aed24be8

**Pull request closed without merging**